ui: clusterviz: fix flicker on location data refresh

vilterp opened the pull request:
Refreshing location data would cause the node map to switch to circle layout for the duration of the request, causing a flicker. This is because the location selector bailed out if the cached reducer state was invalid. Changed it to bail out only if the data doesn't exist.

This flicker was harder to notice because the location data was only refreshed once every ten minutes. I changed it to be every ten seconds like the rest of the reducers. This may be overkill most of the time, but is also nice when setting up your config.

Release note (admin ui change): Fix another flicker in the cluster viz as data was being reloaded.

Commit:
ui: clusterviz: fix flicker and refresh locations more frequently

Refreshing location data would cause the node map to switch to circle
layout for the duration of the request, causing a flicker. This is
because the location selector bailed out if the cached reducer state was
invalid. Changed it to bail out only if the data doesn't exist.

This flicker was harder to notice because the location data was only
refreshed once every ten minutes. I changed it to be every ten seconds
like the rest of the reducers. This may be overkill most of the time,
but is also nice when setting up your config.

Release note (admin ui change): Fix another flicker in the cluster viz
as data was being reloaded.

couchand commented:
On the one hand, yes it's nice to see updates soon when setting up the locations.  On the other hand, it's unfortunately to add yet another noisy request using up bandwidth to reload data that changes very infrequently.  I'd advocate against changing the timeout, instead suggesting users just refresh the page if they're expecting the data is updated, and in a future release work on some kind of push notification for infrequently-changing data.

couchand commented on pkg/ui/src/redux/locations.ts:
```diff
@@ -14,7 +14,7 @@ export function selectLocationsRequestStatus(state: LocationState) {
 
 export function selectLocations(state: LocationState) {
-  if (!state.cachedData.locations.valid) {
+  if (!state.cachedData.locations.data) {
```
I would expect that the natural consequence of this strategy is that all the references to the `select...Status` type selectors should be removed.  It would seem that we never actually care about the `valid` prop the way things are currently set up.

vilterp replied:
Yeah. Will leave to part of a broader refactor around indicating loading states.

couchand replied:
:+1:

Commit:
build: ui: in tests, redirect console.logs to stdout

I find it pretty much impossible to debug a test if I can't console.log
from my code. This makes them show up in Karma's stdout.

Release note: None

vilterp reviewed (commented):
Changed the timeout back. PTAL.

vilterp commented on pkg/ui/karma.conf.js:
```diff
@@ -13,6 +13,14 @@ module.exports = function(config) {
+      level: "log",
+      terminal: true,
+    },
+
```
sneaking this in in its own commit. is there an obvious way of debugging without this that I'm missing?

couchand replied:
this is good thanks

couchand commented on pkg/ui/src/redux/locations.spec.ts:
```diff
@@ -62,22 +62,22 @@ describe("selectLocations", function() {
     assert.deepEqual(selectLocations(state), []);
   });
 
-  it("returns an empty array if location data is invalid", function() {
```
undefined but no data plz

vilterp replied:
added
